randovania/games/prime1/exporter/patch_data_factory.py

```diff
@@ -755,6 +755,26 @@ def create_game_specific_data(self) -> dict:
                 for area in region.areas:
                     level_data[region.name]["rooms"][area.name]["submerge"] = self.rng.random() < probability
 
+        # Replace vanilla missile blast shields with the new ones
+        if not self.configuration.legacy_mode:
+            for region in regions:
+                for area in region.areas:
+                    for node in area.nodes:
+                        if isinstance(node, DockNode) and node.dock_type not in elevator_dock_types:
+                            if str(node.default_dock_weakness) != "Missile Blast Shield (randomprime)":
+                                continue
+
+                            dock_node_key_value = node.extra.get("dock_index")
+                            dock_num = str(dock_node_key_value)
+                            world_name = region.name
+                            room_name = area.name
+
+                            if dock_num not in level_data[world_name]["rooms"][room_name]["doors"]:
+                                level_data[world_name]["rooms"][room_name]["doors"][dock_num] = {}
+
+                            level_data[world_name]["rooms"][room_name]["doors"][dock_num]["shieldType"] = "Blue"
+                            level_data[world_name]["rooms"][room_name]["doors"][dock_num]["blastShieldType"] = "Missile"
```

Comment:
this is a mess.
Make a variable for either `level_data[world_name]["rooms"][room_name]["doors"]` or `level_data[world_name]["rooms"][room_name]["doors"][dock_num]`

Reply:
Used a variable for level_data[world_name]["rooms"][room_name]["doors"]